importer/helpers/helpers.go

```diff
@@ -113,8 +113,9 @@ func (n *UnixfsNode) AddChild(child *UnixfsNode, db *DagBuilderHelper) error {
 	}
 
 	// Pin the child node indirectly
-	if db.mp != nil {
-		db.mp.PinWithMode(childkey, pin.Indirect)
+	err = db.bcb(childkey, false)
```

Comment:
@wking 

>  Couldn't we just handle root-level changes in the caller?

Yes, but @whyrusleeping's point with this PR is to not do things at the caller / per-root because it's very expensive (and unsafe) to pin everything after the fact.  (or do you mean something different?)

Maybe mark-and-sweep fixes this.